compare-evidence-strings/compare.sh

```diff
@@ -1,130 +1,210 @@
 #!/bin/bash
 # A script to compare evidence strings. Please see README.md for details on using it.
 
-# A function to sort key in evidence strings. This makes them more readable and helps comparison through word diff.
-# Also removes the "validated_aginst_schema_version" field entirely, because it frequently changes between the
-# versions, and this change is not important.
+
+
+########################################################################################################################
+echo "Defining functions"
+
+# A function to sort keys in the evidence strings. This makes them more readable and helps comparison through word diff.
+# Also removes several version- and date-related fields which are changed frequently, but do not reflect actual change:
+# * validated_aginst_schema_version
+# * date_asserted
 # The two arguments are input and output JSON files.
 sort_keys () {
   ./jq -S "." --tab <"$1" \
     | tr -d '\t\n' \
     | sed -e 's|}{|}~{|g' \
     | tr '~' '\n' \
     | sed -e 's|,"validated_against_schema_version": "[0-9.]*"||g' \
+    | sed -e 's|"date_asserted": ".\{19\}",||g' \
   > "$2"
 }
 
-# A function to extract some fields from the evidence strings. The fields being extracted are:
+# A function to extract all unique identifying fields from the evidence strings. The fields being extracted are:
 # * ClinVar RCV accession
+# * Phenotype
+# * Allele origin
 # * Variant ID (rsID or, if absent, RCV accession)
 # * Ensembl gene ID
-# * Functional consequence SO code
 extract_fields () {
   ./jq '
     .unique_association_fields.clinvarAccession + "|" +
+    .unique_association_fields.phenotype + "|" +
+    .unique_association_fields.alleleOrigin + "|" +
     .unique_association_fields.variant_id + "|" +
-    .unique_association_fields.gene + "|" +
-    .evidence.gene2variant.functional_consequence
-  ' < "$1" \
-  | tr -d '"' | tr '|' '\t' \
+    .unique_association_fields.gene
+  ' < "$1" | tr -d '"' > "$2"
+}
+
+# Computes a word diff between two files using git diff
+compute_git_diff () {
+  # The --no-index option is important, because otherwise git will refuse to compare the files if you're running this
+  # script from right inside the repository (because the files are untracked).
+  git diff \
+  --minimal \
+  -U0 \
+  --color=always \
+  --word-diff=color \
+  --no-index \
+  "$1" "$2"
+}
+
+# Extract only the functional consequence from the evidence string
+extract_functional_consequences () {
+  ./jq '.evidence.gene2variant.functional_consequence' \
+  | tr -d '"' \
   | sed -e 's|http://purl.obolibrary.org/obo/||g' \
-        -e 's|http://identifiers.org/clinvar.record/||g' \
-        -e 's|http://identifiers.org/dbsnp/||g' \
-        -e 's|http://www.ncbi.nlm.nih.gov/clinvar/||g' \
-  > "$2"
+        -e 's|http://targetvalidation.org/sequence/||g'
```

Comment:
Are these set of URLs a moving target?

Reply:
Not really, no — the change happened simply because I modified the script and now different fields are extracted. For the list of fields which are exported currently, the list of URLs is fixed.